runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2816,3 +2833,162 @@ JITaaSHelpers::getRemoteROMClass(J9Class *clazz, JITaaS::J9ServerStream *stream,
    *classInfoTuple = std::get<0>(recv);
    return romClassFromString(std::get<0>(*classInfoTuple), trMemory->trPersistentMemory());
    }
+
+bool
+JITaaSHelpers::getAndCacheRAMClassInfo(J9Class *clazz, ClientSessionData *clientSessionData, JITaaS::J9ServerStream *stream, ClassInfoDataType dataType, void *data)
+   {
+   JITaaSHelpers::ClassInfoTuple classInfoTuple;
+   ClientSessionData::ClassInfo classInfo;
+   if (!clazz)
```

Comment:
I have see calls being made with the class = NULL that was the reason why I was seeing crash. if you check the function TR_J9VMBase::isClassInitialized you will find the same check for class == NULL.

The circumstance under which this will happen I don't know. If you want I can debug further but since the code is already there is good chances of being some valid reason.

Reply:
If only `TR_J9VMBase::isClassInitialized` has this behavior, then we could place the check in that function alone.

Reply:
also TR_J9VMBase::getByteOffsetToLockword.
If we are targeting function by function then if in future the JIT changes its usage like any other function with NULL class pointer, then JITaaS will crash for that unless we make the appropriate change.

Reply:
ok. If it's more than one place then the code should stay the way it is.